pkg/log/checkpoint/files_test.go

```diff
@@ -0,0 +1,128 @@
+package checkpoint
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/kolide/kit/ulid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFilesFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	var tests = []struct {
+		dirsToCreate int
+		filesPerDir  int
+	}{
+		{dirsToCreate: 2, filesPerDir: 2},
+		{dirsToCreate: 0, filesPerDir: 0},
+	}
+
+	for _, tt := range tests {
+		t.Run("testFilesFound", func(t *testing.T) {
+			dirs, expectedPaths, err := createTestFiles(tempDir, tt.dirsToCreate, tt.filesPerDir)
+			require.NoError(t, err, "creating test files")
+
+			foundPaths := fileNamesInDirs(dirs...)
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
+		})
+	}
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
```

Comment:
I think better to do this as a defer above. Doing this here means it won't trigger if a test fails. Also, if os.RemoveAll fails, no point in failing the testing.

Reply:
updated